sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/misc.scala

```diff
@@ -47,4 +48,69 @@ case class Md5(child: Expression)
       "org.apache.spark.unsafe.types.UTF8String.fromString" +
         s"(org.apache.commons.codec.digest.DigestUtils.md5Hex($c))")
   }
+
+}
+
+/**
+ * A function that calculates an SHA-1 digest and returns it as a hex string
+ * For input of type [[BinaryType]]
+ */
+case class Sha(child: Expression)
+  extends UnaryExpression with ExpectsInputTypes {
+
+  override def dataType: DataType = StringType
+
+  override def expectedChildTypes: Seq[DataType] = Seq(BinaryType, StringType)
```

Comment:
@tarekauel Sorry, I was wrong, didn't figure out the difference between `checkInputDataTypes` and `expectedChildTypes`  of `ExpectsInputTypes`.

In order to get better error message, we should use `checkInputDataTypes`.

`expectedChildTypes` is used to widen the types we support.

For this case, we could use `Seq(BinaryType)` as `expectedChildTypes`, which means it will also support `StringType`, because `StringType` could be casted to BinaryType (which is cheap).

In `eval` and `genCode`, you can expect that child.dataType is BinaryType.